### PR TITLE
feat: add Shadow DOM support for MFE compatibility

### PR DESCRIPTION
**Pull request by jaggaroshu**
Add ShadowAware provider component that uses styled-components' StyleSheetManager to inject styles into a Shadow Root instead of document.head. This enables blend-design-system components to work correctly inside Micro-Frontend (MFE) architectures that use Shadow DOM for style isolation.

- Add ShadowAware component with optional shadowRoot prop
- Add useShadowRoot hook to access shadow root context
- Graceful fallback when shadowRoot is not provided
- Full backward compatibility - no breaking changes

🤖 Generated with [Claude Code](https://claude.ai/code)

### Summary

https://spaces.xyne.juspay.net/chat/dir/cmkny62v402ohjja5if5xnxc6/4bba7cef-3d9d-4fe4-9849-1fbbee50b9b3#origin=4bba7cef-3d9d-4fe4-9849-1fbbee50b9b3&messageId=5c9befa9-bfb2-4f8a-ad0b-40cc85fc6335

### CODE REVIEW

**Comment by vinitkhandal717** [comment]
@jaggaroshu  please fix the checks failure and create a github issue and add it here

**Comment by jagguji** [comment]
https://github.com/juspay/blend-design-system/issues/1267

**Comment by vinitkhandal717** [comment]
Please fix a few edge cases before merge: 
 
(1) Testing Library doesn’t pierce Shadow DOM, so screen.findByRole tests will fail once portals render inside shadow root—tests should query within(target) / container instead. 
 
(2) useDropdownInteractionLock uses a global styleInjected boolean; this breaks when you have multiple shadow roots / main DOM + shadow DOM—track injected per container (Set/WeakSet). 
 
(3) target?.parentNode as insertion container is fragile; use target.getRootNode() and inject into ShadowRoot or document.head deterministically. 

(4) selector scoping differs between shadow vs non-shadow; consider consistent scoping.

File: packages/blend/lib/context/ThemeProvider.tsx [hunk 1]
```diff
@@ -4,12 +4,20 @@ import initTokens from './initComponentTokens'
...
     componentTokens?: ComponentTokenType
     breakpoints?: BreakpointType
     theme?: Theme | string
     children: React.ReactNode
+    /**
+     * Target element for styled-components to inject styles into.
+     * Use this when rendering inside a Shadow DOM.
+     * The element should be inside a shadow root.
+     */
+    target?: HTMLElement
```
[the remark this answers is not in the document]
Comment (jagguji) [reply packages/blend/lib/context/ThemeProvider.tsx:23]: target is only required, as we don't want it to autoDetect the shadowDOm

**Review by vinitkhandal717 — APPROVED** [review]
